spec/requests/api/v1/invoices_request_spec.rb

```diff
@@ -16,22 +16,155 @@
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["status"]).to eq("shipped")
+  end
+
+  it "can find single invoice by customer_id" do
+    get "/api/v1/invoices/find?customer_id=#{@customer.id}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["customer_id"]).to eq(@customer.id)
+  end
+
+  it "can find single invoice by merchant_id" do
+    get "/api/v1/invoices/find?merchant_id=#{@merchant.id}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["merchant_id"]).to eq(@merchant.id)
+  end
+
+  it "can find single invoice by created_at" do
+    get "/api/v1/invoices/find?created_at=#{@invoice.created_at}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["id"]).to eq(@invoice.id)
+  end
+
+  it "can find single invoice by updated_at" do
+    get "/api/v1/invoices/find?updated_at=#{@invoice.updated_at}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["id"]).to eq(@invoice.id)
+  end
+
+  it "can find_all invoice by id" do
+    get "/api/v1/invoices/find_all?id=#{@invoice2.id}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response.count).to eq(1)
+  end
+
+  it "can find_all invoice by status" do
+    get "/api/v1/invoices/find_all?status=shipped"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response.count).to eq(2)
+  end
+
+  it "can find_all invoice by customer_id" do
+    get "/api/v1/invoices/find_all?customer_id=#{@customer.id}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response.count).to eq(2)
+  end
+
+  it "can find_all invoice by merchant_id" do
+    get "/api/v1/invoices/find_all?merchant_id=#{@merchant.id}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response.count).to eq(2)
+  end
+
+  it "can find_all invoice by created_at" do
+    get "/api/v1/invoices/find_all?created_at=#{@invoice.created_at}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response.count).to eq(2)
+  end
+
+  it "can find_all invoice by updated_at" do
+    get "/api/v1/invoices/find_all?updated_at=#{@invoice.updated_at}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response.count).to eq(2)
+  end
+
+  it "can find a random invoice" do
+    get "/api/v1/invoices/random"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response).to_not be_empty
```

Comment:
We could probably do some have_key stuff here, instead of just checking if the response isn't empty.